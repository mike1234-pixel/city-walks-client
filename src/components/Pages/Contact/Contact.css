/** @format */
.contact-grid {
  display: grid;
  grid-template-columns: 1fr 2fr;
  grid-gap: 30px;
}
@media (max-width: 767px) {
  .contact-grid {
    grid-template-columns: repeat(1, 1fr);
  }
}

.contact-main-img img {
  width: 100%;
}

.contact-form textarea {
  min-height: 118px;
}

#contact-submit-btn {
  float: right;
}
@media (max-width: 767px) {
  #contact-submit-btn {
    width: 100%;
    padding: 10px;
    margin-left: 0;
    margin-right: 0;
  }
}

/*# sourceMappingURL=Contact.css.map */
