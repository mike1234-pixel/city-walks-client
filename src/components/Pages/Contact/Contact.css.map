{"version":3,"sourceRoot":"","sources":["Contact.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;EACA;;AAEA;EALF;IAMI;;;;AAKF;EACE;;;AAKF;EACE;;;AAIJ;EACE;;AAEA;EAHF;IAII;IACA;IACA;IACA","file":"Contact.css"}