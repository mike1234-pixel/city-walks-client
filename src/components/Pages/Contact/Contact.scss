/** @format */

#contact-submit-btn {
  @media (max-width: 767px) {
    width: 100%;
    padding: 10px;
    margin-left: 0;
    margin-right: 0;
  }
}
