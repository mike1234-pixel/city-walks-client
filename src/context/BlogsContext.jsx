import { createContext, useState } from "react"

export const BlogsContext = createContext();

export const BlogsContextProvider = (props) => {

    const [blogPosts, setBlogPosts] = useState([])
    const [blogsLoading, setBlogsLoading] = useState(true)

    return (
        <BlogsContext.Provider 
            value={{
                blogPosts,
                setBlogPosts,
                blogsLoading,
                setBlogsLoading,
            }}
        >
            {props.children}
        </BlogsContext.Provider>
    )
}