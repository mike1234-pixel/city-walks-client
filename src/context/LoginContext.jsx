import { createContext, useState } from "react"
import axios from "axios"
import qs from "qs"
import LoginForm from "../components/pages/LoginPage/LoginForm/LoginForm";
import RegistrationForm from "../components/pages/LoginPage/RegistrationForm/RegistrationForm"
import VerificationForm from "../components/pages/LoginPage/VerificationForm/VerificationForm";
import ResetPasswordForm from "../components/pages/LoginPage/ResetPasswordForm/ResetPasswordForm";
import ForgotPasswordForm from "../components/pages/LoginPage/ForgotPasswordForm/ForgotPasswordForm";
import Boards from "../components/pages/LoginPage/LoggedInView/Boards"

export const LoginContext = createContext();

export const LoginContextProvider = (props) => {
    const [popupVisible, setPopupVisible] = useState(false)

    const [firstName, setFirstName] = useState("")
    const [lastName, setLastName] = useState("")
    const [userId, setUserId] = useState("")
    const [registrationEmail, setRegistrationEmail] = useState("")
    const [registrationPassword, setRegistrationPassword] = useState("")
  
    const [loginEmail, setLoginEmail] = useState("")
    const [loginPassword, setLoginPassword] = useState("")
  
    // save all of this in local storage
    const [loggedIn, setLoggedIn] = useState(false)
    const [userFirstName, setUserFirstName] = useState("")
    const [userLastName, setUserLastName] = useState("")
  
    const [verificationEmail, setVerificationEmail] = useState("")

    const [resetPasswordEmail, setResetPasswordEmail] = useState("")
    const [resetPasswordOldPassword, setResetPasswordOldPassword] = useState("")
    const [resetPasswordNewPassword, setResetPasswordNewPassword] = useState("")
    const [resetPasswordConfirmNewPassword, setResetPasswordConfirmNewPassword] = useState("")

    const [forgotPasswordEmail, setForgotPasswordEmail] = useState("")

    const handlePopup = () => {
      setPopupVisible(false)
      localStorage.setItem("popupVisible", false)
    }

    // set display form on LoginPage
    const [form, setForm] = useState("loggedOutView");

    let displayForm = <Boards/>;
    if (form === "boards") {
      displayForm = <Boards/>;
    } else if (form === "loginForm") {
      displayForm = <LoginForm />;
    } else if (form === "registrationForm") {
      displayForm = <RegistrationForm />;
    } else if (form === "verificationForm") {
      displayForm = <VerificationForm />;
    } else if (form === "resetPasswordForm") {
      displayForm = <ResetPasswordForm />;
    } else if (form === "forgotPasswordForm") {
      displayForm = <ForgotPasswordForm />;
    }

    const handleChangeRegistration = (event) => {
        switch(event.target.name) {
            case "registration-fname":
              setFirstName(event.target.value)
              break;
            case "registration-lname":
              setLastName(event.target.value)
              break;
            case "registration-email":
              setRegistrationEmail(event.target.value)
              break;
            case "registration-password":
              setRegistrationPassword(event.target.value)
              break;
          } 
    }
  
    const handleSubmitRegistration = (event) => {
      console.log("handle submit triggered")
        event.preventDefault()
  
        const payload = {
            fname: firstName,
            lname: lastName,
            email: registrationEmail,
            password: registrationPassword,
          };
  
    axios
        .post("http://localhost:5000/register-user", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "An account with this email already exists.") {
            alert("An account with this email already exists.")
          } else if (res.data === "We have sent you an email. Please verify your account by clicking the link in the mail.") {
            alert("We have sent you an email. Please verify your account by clicking the link in the email. (This code expires after 10 minutes)")
           } else {
            alert("Registration Complete")
            setLoggedIn(true)
            setFirstName("")
            setLastName("")
            setRegistrationEmail("")
            setRegistrationPassword("")
            setUserFirstName(res.data.fname)
            setUserLastName(res.data.lname)
            window.scrollTo(0, 0)
          }
        });
    }
  
    const handleChangeLogin = (event) => {
      switch(event.target.name) {
          case "login-email":
            setLoginEmail(event.target.value)
            break;
          case "login-password":
            setLoginPassword(event.target.value)
            break;
        } 
  }
  
  const handleSubmitLogin = (event) => {
    console.log("handle submit login triggered")
      event.preventDefault()
  
      const payload = {
          email: loginEmail,
          password: loginPassword
        };
  
  axios
      .post("http://localhost:5000/login-user", qs.stringify(payload))
      .then((res, err) => {
        if (err) {
          console.log(err);
        } else if (res.data === "Your account exists but is not activated. Please click 'verify account' for email verification.") {
          alert("Your account exists but is not activated. Please click 'verify account' for email verification.")
        } else if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Login Complete")
          setLoggedIn(true)
          setLoginEmail("")
          setLoginPassword("")
          setForm("")
          setUserId(res.data._id)
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          localStorage.setItem("loggedIn", true)
          localStorage.setItem("userId", res.data._id)
          localStorage.setItem("userFirstName", res.data.fname)
          window.scrollTo(0, 0)
        }
      });
  }

  // resend verification email

  const handleChangeVerification = (event) => {
    switch(event.target.name) {
        case "verification-email":
          setVerificationEmail(event.target.value)
          break;
      }   
  }

const handleSubmitVerification = (event) => {
    console.log("handle submit verification")
    event.preventDefault()

    const payload = {
        email: verificationEmail,
      };

axios
    .post("http://localhost:5000/reverify-user", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("Verification email sent. Check your inbox.")
        setVerificationEmail("")
        window.scrollTo(0, 0)
      }
    })
}

  // reset password

  const handleChangeResetPassword = (event) => {
    switch(event.target.name) {
        case "reset-email":
          setResetPasswordEmail(event.target.value)
          break;
        case "old-password":
          setResetPasswordOldPassword(event.target.value)
          break;
        case "new-password":
          setResetPasswordNewPassword(event.target.value)
          break;
        case "confirm-new-password":
          setResetPasswordConfirmNewPassword(event.target.value)
          break;
      }   
  }

const handleSubmitResetPassword = (event) => {
    console.log("handle submit reset password")
    event.preventDefault()

    const payload = {
        email: resetPasswordEmail,
        oldPassword: resetPasswordOldPassword,
        newPassword: resetPasswordNewPassword,
      };

      if (resetPasswordNewPassword === resetPasswordConfirmNewPassword) {
        axios
        .post("http://localhost:5000/reset-password-with-old-password", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "user not found") {
            alert("We could not find your account. Please try again.")
          } else if (res.data === "old password does not match password in the database") {
            alert("The password you entered does not match the account. Please try again or use the Forgot Password option.")
          } else if (res.data === "password successfully updated") {
            alert("Password reset. You can now login.")
            setResetPasswordEmail("")
            setResetPasswordOldPassword("")
            setResetPasswordNewPassword("")
            setResetPasswordConfirmNewPassword("")
            window.scrollTo(0, 0)
          }
        })
      } else {
        alert("Passwords don't match. Please try again.")
      }

}

  // forgot password

  const handleChangeForgotPassword = (event) => {
    switch(event.target.name) {
        case "forgot-password-email":
          setForgotPasswordEmail(event.target.value)
          break;
      }   
  }

const handleSubmitForgotPassword = (event) => {
    console.log("handle submit forgot password")
    event.preventDefault()

    const payload = {
        email: forgotPasswordEmail,
      };

axios
    .post("http://localhost:5000/forgot-password", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("We have sent you an email. Please click the click in your email to reset your password")
        setForgotPasswordEmail("")
        window.scrollTo(0, 0)
      }
    })
}
  
  const logOut = () => {
    localStorage.clear()
    localStorage.setItem("popupVisible", false)
    setLoggedIn(false)
    setUserId("")
    setUserFirstName("")
    setUserLastName("")
    setForm("boards")
  }

    return (
        <LoginContext.Provider 
            value={{
              popupVisible,
              setPopupVisible,
              handlePopup,
               // display form
               displayForm,
               setForm,
               // logged in state 
               loggedIn,
               // registration state and functions
                firstName,
                lastName,
                registrationEmail,
                registrationPassword,
                handleChangeRegistration,
                handleSubmitRegistration,
                // login state and functions
                loginEmail,
                loginPassword,
                handleChangeLogin: handleChangeLogin,
                handleSubmitLogin: handleSubmitLogin,
                setLoggedIn,
                setUserFirstName,
                setUserId,
                // logged in user data
                userFirstName,
                userId,
                userLastName,
                // resend verification email
                verificationEmail,
                handleChangeVerification,
                handleSubmitVerification,
                // reset password
                resetPasswordEmail,
                resetPasswordOldPassword,
                resetPasswordNewPassword,
                resetPasswordConfirmNewPassword,
                handleChangeResetPassword,
                handleSubmitResetPassword,
                // forgot password
                forgotPasswordEmail,
                handleChangeForgotPassword,
                handleSubmitForgotPassword,
                // logout function
                logOut
            }}
        >
            {props.children}
        </LoginContext.Provider>
    )
}

// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();

