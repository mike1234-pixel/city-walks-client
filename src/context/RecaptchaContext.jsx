import { createContext, useEffect } from "react"

export const RecaptchaContext = createContext();

export const RecaptchaContextProvider = (props) => {
    
  const siteKey = "6LdmxiUaAAAAAIYySt3c8XvwOMokTQ_SW2cYkvMw";
 
  // dynamically load the javascript api using the site key
    useEffect(() => {
      const loadScriptByURL = (id, url, callback) => {
        const isScriptExist = document.getElementById(id);
    
        if (!isScriptExist) {
          var script = document.createElement("script");
          script.type = "text/javascript";
          script.src = url;
          script.id = id;
          script.onload = function () {
            if (callback) callback();
          };
          document.body.appendChild(script);
        }
    
        if (isScriptExist && callback) callback();
      }
    
      // load the script by passing the URL
      loadScriptByURL("recaptcha-key", `https://www.google.com/recaptcha/api.js?render=${siteKey}`, () => {
        console.log("recaptcha script loaded!");
      });
    }, []);

    return (
        <RecaptchaContext.Provider 
            value={{
            siteKey
            }}>
            {props.children}
        </RecaptchaContext.Provider>
    )
}