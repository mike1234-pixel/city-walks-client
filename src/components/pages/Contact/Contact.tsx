import React from 'react'
import { useState, useEffect } from "react"
import { MDBInput, MDBBtn, MDBIcon, MDBContainer } from "mdbreact"
import axios, { AxiosError, AxiosResponse } from "axios"
import qs from "qs"
import { connect } from 'react-redux'
import Message from '../../../types/PostRequests/Message'
import RootState from '../../../types/State/Root/State'
import "./Contact.scss"

interface Props {
  sitekey: string;
}

const Contact: React.FC<Props> = (props: Props) => {

  const { sitekey } = props

  const [contactName, setContactName] = useState<string>("");
  const [contactEmail, setContactEmail] = useState<string>("");
  const [contactMessage, setContactMessage] = useState<string>("");

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "contact-name":
        setContactName(event.target.value);
        break;
      case "contact-email":
        setContactEmail(event.target.value);
        break;
      case "contact-message":
        setContactMessage(event.target.value);
        break;
    }
  };

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault();

    window.grecaptcha.ready(() => {
      window.grecaptcha.execute(sitekey, { action: 'submit' }).then((token: string) => {
        submitData(token);
      });
    });

    const submitData = (token: string) => {

      const payload: Message = {
        name: contactName,
        email: contactEmail,
        message: contactMessage,
        gRecaptchaResponse: token
      };

      axios
        .post("https://city-walks.herokuapp.com/contact-form", qs.stringify(payload))
        .then((res: AxiosResponse) => {
          if (res.data === "request failed recaptcha") {
            alert("request failed recaptcha.")
          } else {
            alert("Thanks for getting in touch. We will be in touch shortly.");
            setContactName("");
            setContactEmail("");
            setContactMessage("");
            window.scrollTo(0, 0);
          }
        }).catch((err: AxiosError) => {
          console.log(err)
        });
    };
  }

  return (
    <MDBContainer>
      <div>
        <div className="page-heading-container min-page-height">
          <h1 className="page-heading">Contact</h1>
          <form onSubmit={handleSubmit} className="contact-form display-form">
            <MDBInput
              key="input-5"
              type="text"
              name="contact-name"
              id="contact-name"
              value={contactName}
              label="name"
              onChange={handleChange}
              required
            />
            <MDBInput
              key="input-6"
              type="email"
              name="contact-email"
              id="contact-email"
              value={contactEmail}
              label="email"
              onChange={handleChange}
              required
            />
            <MDBInput
              key="input-7"
              type="textarea"
              rows="5"
              name="contact-message"
              id="contact-message"
              value={contactMessage}
              label="message"
              onChange={handleChange}
              required
            />
            <MDBBtn outline color="elegant" type="submit" id="contact-submit-btn">
              Send Message <MDBIcon far icon="paper-plane" />
            </MDBBtn>
          </form>
        </div>
      </div>
    </MDBContainer>
  );
};

const mapStateToProps: (state: RootState) => void = (state) => ({
  sitekey: state.recaptchaState.sitekey,
});

export default connect(mapStateToProps)(Contact);
