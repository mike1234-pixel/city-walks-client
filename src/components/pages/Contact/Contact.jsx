/** @format */

import { useState, useContext, useEffect } from "react"
import { RecaptchaContext } from "../../../context/RecaptchaContext"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact";
import PopUp from "../../PopUp/PopUp"
import { LoginContext } from "../../../context/LoginContext"
import axios from "axios";
import qs from "qs"
import "./Contact.css";

const Contact = () => {

  const { popupVisible } = useContext(LoginContext)

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const [contactName, setContactName] = useState("");
  const [contactEmail, setContactEmail] = useState("");
  const [contactMessage, setContactMessage] = useState("");

  const {siteKey} = useContext(RecaptchaContext)

  const handleChange = (event) => {
    switch (event.target.name) {
      case "contact-name":
        setContactName(event.target.value);
        break;
      case "contact-email":
        setContactEmail(event.target.value);
        break;
      case "contact-message":
        setContactMessage(event.target.value);
        break;
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    window.grecaptcha.ready(() => {
      window.grecaptcha.execute(siteKey, { action: 'submit' }).then(token => {
        submitData(token);
      });
    });

    const submitData = token => {

      const payload = {
        name: contactName,
        email: contactEmail,
        message: contactMessage,
        gRecaptchaResponse: token
      };

    axios
      .post("http://localhost:5000/contact-form", qs.stringify(payload))
      .then((res, err) => {
        if (err) {
          console.log(err);
        } if (res.data === "request failed recaptcha") {
          alert("request failed recaptcha.")
        } else {
          alert("Thanks for getting in touch. We will be in touch shortly.");
          setContactName("");
          setContactEmail("");
          setContactMessage("");
          window.scrollTo(0, 0);
        }
      });
  };
}

  return (
  <div>
  {popupVisible && <PopUp/>}
    <div className="page-heading-container min-page-height">
      <h1 className="page-heading">Contact</h1>
      <form onSubmit={handleSubmit} className="contact-form display-form">
        <MDBInput
          key="input-5"
          type="text"
          name="contact-name"
          id="contact-name"
          value={contactName}
          label="name"
          onChange={handleChange}
          required
        />
        <MDBInput
          key="input-6"
          type="email"
          name="contact-email"
          id="contact-email"
          value={contactEmail}
          label="email"
          onChange={handleChange}
          required
        />
        <MDBInput
          key="input-7"
          type="textarea"
          rows="5"
          name="contact-message"
          id="contact-message"
          value={contactMessage}
          label="message"
          onChange={handleChange}
          required
        />
        <MDBBtn outline color="elegant" type="submit">
          Send Message <MDBIcon far icon="paper-plane" />
        </MDBBtn>
      </form>
    </div>
  </div>
  );
};

export default Contact;
