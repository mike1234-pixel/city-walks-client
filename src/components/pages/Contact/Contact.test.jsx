import renderer from 'react-test-renderer'
import { RecaptchaContextProvider } from "../../../context/RecaptchaContext"
import { LoginContextProvider } from "../../../context/LoginContext"
import Contact from './Contact'

// snapshot test
it('Contact Page renders correctly', () => {  

  const tree = renderer
    .create(
      <RecaptchaContextProvider>
        <LoginContextProvider>
          <Contact/>
        </LoginContextProvider>
      </RecaptchaContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});