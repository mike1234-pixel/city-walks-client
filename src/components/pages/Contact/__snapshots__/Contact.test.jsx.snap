// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page renders correctly 1`] = `
<div>
  <div
    className="page-heading-container min-page-height"
  >
    <h1
      className="page-heading"
    >
      Contact
    </h1>
    <form
      className="contact-form display-form"
      onSubmit={[Function]}
    >
      <div
        className="md-form"
      >
        
        <input
          aria-disabled={false}
          className="form-control"
          data-test="input"
          disabled={false}
          id="contact-name"
          name="contact-name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          aria-labelledby=""
          className=""
          data-error=""
          data-success=""
          htmlFor="contact-name"
          id=""
          onClick={[Function]}
        >
          name
        </label>
      </div>
      <div
        className="md-form"
      >
        
        <input
          aria-disabled={false}
          className="form-control"
          data-test="input"
          disabled={false}
          id="contact-email"
          name="contact-email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          required={true}
          type="email"
          value=""
        />
        <label
          aria-labelledby=""
          className=""
          data-error=""
          data-success=""
          htmlFor="contact-email"
          id=""
          onClick={[Function]}
        >
          email
        </label>
      </div>
      <div
        className="md-form"
      >
        
        <textarea
          aria-disabled={false}
          className="md-textarea form-control"
          data-test="input"
          disabled={false}
          id="contact-message"
          name="contact-message"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          required={true}
          rows="5"
          value=""
        />
        <label
          aria-labelledby=""
          className=""
          data-error=""
          data-success=""
          htmlFor="contact-message"
          id=""
          onClick={[Function]}
        >
          message
        </label>
      </div>
      <button
        className="btn-elegant btn-outline-elegant btn Ripple-parent"
        data-test="button"
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        type="submit"
      >
        Send Message 
        <i
          className="far fa-paper-plane"
          data-test="fa"
        />
        <div
          className="Ripple Ripple-outline "
          data-test="waves"
          style={
            Object {
              "height": "0px",
              "left": "0px",
              "top": "0px",
              "width": "0px",
            }
          }
        />
      </button>
    </form>
  </div>
</div>
`;
